app/services/supplementary-billing/reissue-invoices.service.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+/**
+ * Handles the reissuing of invoices
+ * @module ReissueInvoicesService
+ */
+
+const BillingInvoiceModel = require('../../models/water/billing-invoice.model.js')
+const BillingInvoiceLicenceModel = require('../../models/water/billing-invoice-licence.model.js')
+const BillingTransactionModel = require('../../models/water/billing-transaction.model.js')
+const FetchInvoicesToBeReissuedService = require('./fetch-invoices-to-be-reissued.service.js')
+const ReissueInvoiceService = require('./reissue-invoice.service.js')
+
+/**
+ * Handles the reissuing of invoices
+ *
+ * We receive the billing batch that the reissued invoices are to be created on and infer from this the region to be
+ * reissued. We check this region for invoices marked for reissuing. For each one of these we call
+ * `ReissueInvoiceService` which handles the actual reissuing of an invoice and collects the returned invoice, invoice
+ * licence and transaction data which we batch persist once all invoices have been reissued. Finally we return a boolean
+ * to indicate whether or not any invoices were reissued.
+ *
+ * @param {module:BillingBatchModel} reissueBillingBatch The billing batch that the reissued invoices will be created on
+ *
+ * @returns {Boolean} `true` if any invoices were reissued; `false` if not
+*/
+
+async function go (reissueBillingBatch) {
+  const sourceInvoices = await FetchInvoicesToBeReissuedService.go(reissueBillingBatch.regionId)
+
+  if (sourceInvoices.length === 0) {
+    return false
+  }
+
+  const dataToPersist = {
+    billingInvoices: [],
+    billingInvoiceLicences: [],
+    transactions: []
```

Comment:
I was just wondering why the invoices and invoice-licences keep the `billing` prefix but `transactions` doesn't. Have I let this convention slip through into other areas!? 😰